base_images/Dockerfile.bamboo

```diff
@@ -14,11 +15,22 @@ curl "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/linux_64b
 amazon-linux-extras install epel -y && \
 yum update -y
 
-# install Python 3, pip, CLI utilities, session manager and other tools
+# install pip, CLI utilities, session manager and other tools
 RUN \
-yum install -y jq gcc git make openssl wget zip unzip tar python3-devel nodejs yarn python3-pip awscli session-manager-plugin.rpm procps parallel && \
+yum install -y jq gcc git make openssl wget zip unzip tar nodejs yarn python3-pip awscli session-manager-plugin.rpm procps parallel && \
```

Comment:
Remove `python3-pip` from the list and add the development libraries needed to install python 3.9 `openssl-devel bzip2-devel libffi-devel ncurses-devel sqlite-devel readline-devel uuid-devel libuuid-devel gdbm-devel xz-devel`

I would also install development tools as well to get additional options in Python
`yum groupinstall "Development Tools && \`

Note you will need to build python before running the pip command on line 21. I will add additional notes in your install python block. I would move the entire Python build into this block.

Reply:
updated as stated above. For some reason, I am getting error while running 
`pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org pydoc-markdown`

Got same error as well with this
`pip3.9 install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org pydoc-markdown`


Error:
#6 323.4 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#6 323.5 /bin/sh: pip: command not found

Reply:
looks like we don't even need this pip install comment for running unit tests in bamboo.
All lambas bin/run_tests.sh already have this line of code